react-validated-input-component/src/validated-input.jsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+
+class ValidatedInput extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleChange = this.handleChange.bind(this);
+    this.state = {
+      filled: false,
+      longEnough: false
+    };
+  }
+
+  handleChange(event) {
+    const length = event.target.value.length;
+    if (length >= 8) {
+      this.setState({
+        filled: true,
+        longEnough: true
+      });
+    } else if (length < 8 && length > 0) {
+      this.setState({
+        filled: true,
+        longEnough: false
+      });
+    } else {
+      this.setState({
+        filled: false,
+        longEnough: false
+      });
+    }
```

Comment:
You still need to update the `password` property in state to store the inputs value.

```JSX
password: event.target.value
```